### ISSUE TITLE
Allow the application to run as a service

### ISSUE THREAD

**Issue by tharakadesilva** [tharakadesilva/SlackUserPresenceManager#5]
With this we will be able to have both log on and log off events giving us the ability to add the "computer off" event to the custom status.

**Comment by wcchristian** [comment]
To do this, I have a PR up https://github.com/tharakadesilva/SlackUserPresenceManager/pull/9 that handles adding the code for log-out/shut-down.

To complete this issue, the following tasks must be completed.

- Create this into a service so that it can be set to autorun at startup. Likely use something like jsmooth to build this jar into an exe and then set the exe to run as a service, more investigation will need to be done to complete this.
- Add an if check to the only set the status to auto from the logged off state if slack is running, Since we are auto starting this when Windows starts, we need to not set the status when the service starts, but only when slack is running. We will likely need to build another hook for this.

**Comment by tharakadesilva** [comment]
https://wrapper.tanukisoftware.com/doc/english/qna-service.html might an interesting way to approach the service issue.

The if condition is definitely a good idea. I didn't think of that. That if condition might even have to be included to the other hooks as well.

**Comment by wcchristian** [comment]
@tharakadesilva agreed, I'll look into the tanuki wrapper! I always thought it costed a ton but since we are open source I believe that we can use the community version for free. I'll dig deeper
